### PR TITLE
Add introduction & note plugin activation on package phase

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -8,6 +8,21 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
...
+Maven tooling in Quarkus relies on a plugin that the Quarkus team developed to provide an enjoyable developer experience.
+While we recommend that you <<project-creation,scaffold>> new projects, you can find how to
+<<build-tool-maven,configure the plugin>> manually, below.
+
+Many of the functionality of the plugin are also available in a separate command-line interface tool called `quarkus`.
+
+[NOTE]
+====
+By default, the Quarkus Maven plugin is only triggered on the `package` phase of the build process. It's therefore not
+enough to simply `compile` the project.
```
[the remark this answers is not in the document]
Comment (metacosm) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: Is this documented somewhere? This change was brought about by https://github.com/quarkiverse/quarkus-operator-sdk/issues/272 where people were wondering why their files didn't get generated on `mvn compile`.

[the remark this answers is not in the document]
Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: These goals mentioned in https://quarkus.io/guides/grpc-getting-started, which introduced them in the first place. But they use the ServiceLoader to load `CodeGenProvider` impl from the classpath, so it's possible to re-use these goals for other code generators.

[the remark this answers is not in the document]
Comment (metacosm) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: OK, but it doesn't make sense for someone looking to learn about what the maven plugin can do to go look at the GRPC guide so this should probably be documented in this guide instead. 😄

[the remark this answers is not in the document]
Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: Right

**Comment by gsmet** [comment]
@metacosm can you squash this one? Thanks.

File: docs/src/main/asciidoc/maven-tooling.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (gsmet) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: @aloubyansky not sure what is the conclusion of your discussion? Should this be merged or not?

[the remark this answers is not in the document]
Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc:23]: The phrase is inaccurate. It either has to be re-phrased or simply removed. I am not sure what was meant to be highlighted to make a specific suggestion. E.g. `compile` is not enough for what?

**Comment by gsmet** [comment]
I will close this one. I'm not sure the introduction brings much (and we will probably have the doc team work on this at some point) and the info added is apparently controversial.

**Pull request closed without merging**